docs/guides/4-custom-rulesets.md

```diff
@@ -0,0 +1,160 @@
+# Custom Rulesets
+
+Want to go beyond tweaking a ruleset, and learn how to make your own?
+
+## Adding Rules
+
+Add your own rules under the `rules` property in your `.spectral.yml`, or another ruleset file.
+
+```yaml
+rules:
+  my-rule-name:
+    description: Tags must have a description.
+    given: $.tags[*]
+    severity: error
+    then:
+      field: description
+      function: truthy
+```
+
+Spectral has [built-in functions](../reference/functions.md) such as `truthy` or `pattern`, which can be used to power rules.
+
+By default, Spectral processes each rule on "resolved document" (a file where all `$ref`s have been resolved. If you would like to have an original input supplied to your rule, you can place `resolved` property as follows:
+
+```yaml
+rules:
+  my-rule-name:
+    description: Tags must have a description.
+    given: $.tags[*]
+    severity: error
+    resolved: false
+    then:
+      field: description
+      function: truthy
+```
+
+You might find `resolved` useful if your rule requires access to `$ref` values specifically, for example if you want to enforce conventions on the folder structure used for [splitting up documents](https://stoplight.io/blog/keeping-openapi-dry-and-portable/).
+
+In most cases, you will want to operate on resolved document.
+
+### Given
+
+The `given` property is one of only two on required properties on each rule definition (the other being `then`).
+
+It can be any valid JSONPath expression or an array of JSONPath expressions.
+[JSONPath Online Evaluator](http://jsonpath.com/) is a helpful tool to determine what `given` path you want.
+
+### Severity
+
+The `severity` keyword is optional and can be `error`, `warn`, `info`, or `hint`.
+
+The default value is `warn`.
+
+### Then
+
+The Then part of the rules explains what to do with the `given` JSONPath, and involves two required keywords:
+
+```yaml
+then:
+  field: description
+  function: truthy
+```
+
+The `field` keyword is optional, and is for applying the function to a specific property in an object. If omitted the function will be applied to the entire target of the `given` JSON Path. The value can also be `@key` to apply the rule to a keys of an object.
+
+```yaml
+given: '$.responses'
+then:
+  field: '@key'
+  function: pattern
+  functionOptions:
+    match: '^[0-9]+$'
+```
+
+The above pattern based rule would error on `456avbas` as it is not numeric.
+
+```yaml
+responses:
+  123:
+    foo: bar
+  456avbas:
+    foo: bar
+```
+
+## Extending Rules
+
+When extending another ruleset, you can actually extend and modify rules it has declared by adding a rule to your own ruleset with the same name.
+
+```yaml
+extends: spectral:oas
+rules:
+  my-rule-name:
```

Comment:
@philsturgeon This example doesn't really relate with the previous sentence

Reply:
![image](https://user-images.githubusercontent.com/92363/79757350-45b47a00-831c-11ea-860b-833730bd2c89.png)

@philsturgeon I may be missing something here. The doc says "You can actually extend and modify rules it has declared by adding a rule to your own ruleset with **the same name**.", but `my-rule-name` doesn't exist in the oas ruleset. Am I misunderstanding what this sample demonstrates?

Reply:
Fixed!